book/src/programs.md

```diff
@@ -1,32 +1,67 @@
+## Deploying Programs to a Cluster
 
 <img alt="SDK tools" src="img/sdk-tools.svg" class="center"/>
 
-As shown in the diagram above an untrusted client, creates a program in the
-language of their choice, (i.e. C/C++/Rust/Lua), and compiles it with LLVM to a
-position independent shared object ELF, targeting BPF bytecode, and sends it to
-the Solana cluster. Next, the client sends messages to the Solana cluster,
-which target that program. The Solana runtime loads the previously submitted
-ELF and passes it the client's message for interpretation.
-
-## Persistent Storage
-
-Solana supports several kinds of persistent storage, called *accounts*:
-
-1. Executable
-2. Owned by a client
-3. Owned by a program
-4. Credit-only
-
-All accounts are identified by public keys and may hold arbirary data.
-When the client sends messages to programs, it requests access to storage
-using those keys. The runtime loads the account data and passes it to the
-program. The runtime also ensures accounts aren't written to if not owned
-by the client or program. Any writes to credit-only accounts are discarded
-unless the write was to credit tokens. Any user may credit other accounts
-tokens, regardless of account permission.
+As shown in the diagram above a client creates a program and compiles it to an
+ELF shared object containing BPF bytecode and sends it to the Solana cluster.
+The cluster stores the program locally and makes it available to clients via a
+*program ID*. The program ID is a *public key* generated by the client and is
+used to reference the program in subsequent transactions.
+
+## Interacting with On-chain Programs
+
+After a program has been deployed, a client utilizes it by sending transactions
+to the Solana cluster. A transaction contains one or more *instructions*. Each
+instruction references the program ID that corresponds to the program that the
+runtime should use to execute it. Instructions are executed atomically. If any
+instruction is invalid, any changes made within the transaction are discarded.
+
+## Storing State between Transactions
+
+If the program needs to store state between transactions, it does so using
+*accounts*. Accounts are similar to files in operating systems such as Linux.
+Like files, an account may hold arbitrary data and that data persists beyond
+the lifetime of a program. Also like files, an account includes metadata that
+tells the runtime who is allowed to access the data and how. Unlike files, the
+account includes metadata for the lifetime of the file. That lifetime is
+expressed in "tokens", which is a number of fractional native tokens, called
+*lamports*. Accounts are held in validator memory and pay "rent" to stay there.
+Each fullnode periodically scan all accounts and collects rent. Any account
+that drops to zero lamports is purged.
+
+If an account is marked "executable", it will only be used by a *loader* to run
+programs. For example, programs compiled to BPF are marked executable and
```

Comment:
I think this is a little more correct: "For example, BPF programs are stored in accounts marked executable..."

Reply:
Thanks, but then the reader asks, "wait, what's a BPF program?"

Reply:
I think that's pretty clear from the first paragraph of this doc. But I suppose you could write "BPF-compiled program"

Reply:
Okay, BPF-compiled, I like it, thanks. I also changed it to singular.